tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,16 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
+  for (int i = 0; i < perm.dim_size(0); ++i) {
```

Comment:
Nit: `perm.NumElements()` and saved to a local variable instead of inside the loop

Reply:
Yes.It is more readable. Thanks!